MovingBlocks/TerasologyLauncher
# feat: 1:1 migration from LauncherSettings >>> Settings

## Contains

This does not (yet) make use of any feature we gain by having JavaFX properties or bindings, but on the upside this should be a feature-equivalent migration to the new settings POJO.

I know that the migration is not pretty, but I'd like to improve on the details in separate follow-up PRs to keep the review load bearable.

_Most changes where done with IntelliJ's "Type Migration" and direct migration of getter/setter calls to access of the property value._

## How to test

Ensure that all tests are still passing when running `gradlew test`.

Start the launcher and play around with the settings.

- changed settings should have an effect, e.g., whether to show pre-releases (NOTE: changing the locale only has an effect after restart. This is a known issues). 
- saved settings should be persisted and stay the same between restarts
- no unexpected errors or crashes
- both settings files (Java properties and JSON) are written to the launcher user directory (defaults to `~/.terasologylauncher/`)

---

Based on #666

Contributes to #522
